core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -1569,12 +1569,11 @@ NATable *BindWA::getNATable(CorrName& corrName,
       ((QualifiedName&)(table->getTableName())).setIsVolatile(TRUE);
     }
       
-  // For now, do not allow access through the Trafodion external name created for
-  // the HIVE object unless the inDDL flag is set.  inDDL is set for drop 
-  // table and SHOWDDL statements.  
-  // TDB - may want to merge the Trafodion version with the HIVE version.
-  // TDB - similar operation may be needed for external HBase tables
-  if ((table) && (table->isExternalTable() && (! bindWA->inDDL())))
+  // For now, don't allow access through the Trafodion external name created for
+  // native HIVE or HBASE objects unless the allowExternalTables flag is set.  
+  // allowExternalTables is set for drop table and SHOWDDL statements.  
+  // TDB - may want to merge the Trafodion version with the native version.
+  if ((table) && (table->isExternalTable() && (! bindWA->allowExternalTables())))
```

Comment:
We could open this up and allow the customer to use either name.  The same results should be returned independent of the name.  They are sort of like synonyms. But the way it is coded is that the customer will always use the hive.sch.table or hbase.sch.table name.  They really don't need to worry about the registered (external) name.  

We could open it up for all requests but there will be additional code and testing.  I could create a JIRA to change this or wait and see how people use external tables to see if this is a desired feature.